<?xml version="1.0" encoding="utf-8"?>

<launch>
  <!-- Console launch prefix -->
  <arg name="launch_prefix" default=""/>
  <arg name="node_base_namespace" default="/nepi/dev_id/" />
  <arg name="file_name" default="nepi_pytorch_ros_node.py" />
  <arg name="node_name" default="nepi_pytorch_ros" />


  <!-- Config and weights folder. -->
  <arg name="weights_path"          default="mnt/nepi_storage/ai_models/pytorch/models/weights"/>
  <arg name="config_path"           default="mnt/nepi_storage/ai_models/pytorch/models/cfg"/>
  
  <!-- network parameter files -->
  <arg name="network_param_file"         default="mnt/nepi_storage/ai_models/pytorch/config/common_object_detection_small.yaml"/>

  <!-- Input image -->
  <arg name="source_img_topic"          default="$(find idx)/color_2d_image"/>


  <!-- detector threshold -->
  <arg name="detector_threshold"        default="0.3"/>

  <!-- Load parameters -->
  <rosparam command="load" ns="$(arg node_base_namespace)/$(arg node_name)" file="$(arg network_param_file)"/>

  <!-- Start pytorch and ros wrapper -->
  <node pkg="nepi_pytorch" ns="$(arg node_base_namespace)" type="$(arg file_name)" name="$(arg node_name)" output="screen" launch-prefix="$(arg launch_prefix)">
    <param name="node_base_namespace"   value="$(arg node_base_namespace)" />
    <param name="weights_path"          value="$(arg weights_path)" />
    <param name="config_path"           value="$(arg config_path)" />
    <param name="detector_threshold" value="$(arg detector_threshold)" />
    <param name="source_img_topic" value="$(arg source_img_topic)" />
  </node>

 <!--<node name="republish" type="republish" pkg="image_transport" output="screen" 	args="compressed in:=/front_camera/image_raw raw out:=/camera/image_raw" /> -->
</launch>
